pkg/operator/operator_test.go

```diff
@@ -91,98 +148,53 @@ func TestOperatorSync_NoOp(t *testing.T) {
 
 	for _, tc := range cases {
 		t.Run(string(tc.platform), func(t *testing.T) {
-			infra := &configv1.Infrastructure{
+			infra := &v1.Infrastructure{
 				ObjectMeta: metav1.ObjectMeta{
 					Name: "cluster",
 				},
-				Status: configv1.InfrastructureStatus{
+				Status: v1.InfrastructureStatus{
 					Platform: tc.platform,
 				},
 			}
 
-			configClient := fakeos.NewSimpleClientset()
-			configSharedInformer := configinformersv1.NewSharedInformerFactoryWithOptions(configClient, 10*time.Minute)
-			featureGateInformer := configSharedInformer.Config().V1().FeatureGates()
+			stopCh := make(<-chan struct{})
+			optr := newFakeOperator(nil, []runtime.Object{infra}, stopCh)
+			optr.imagesFile = imagesFilePath
 
-			optr := Operator{
-				eventRecorder:     record.NewFakeRecorder(5),
-				osClient:          fakeconfigclientset.NewSimpleClientset(infra),
-				imagesFile:        imagesFilePath,
-				featureGateLister: featureGateInformer.Lister(),
+			err = optr.sync("test-key")
+			if !assert.NoError(t, err, "unexpected sync failure") {
+				t.Fatal()
 			}
 
-			err = optr.sync("test-key")
-			if !tc.expectedNoop {
```

Comment:
@enxebre IIRC, you authored this unit test. Is it ok to check for `!assert.NoError(t, err, "unexpected sync failure")` no matter what `tc.expectedNoop` is set to?

Reply:
@ingvagabund by the way before the all correct provider's tests failed to deploy, because of missing deployment file, I do not sure if at proper test